### ISSUE TITLE
error Missing an expected key: 'NSPrivacyCollectedDataTypes' prevents generating Apple Privacy Reports

### ISSUE THREAD

**Issue by AleCatSS** [aws-amplify/amplify-swift#3552]
### Describe the bug

Hello,

regarding the [Apple Privacy Manifest file](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api?language=objc), I noticed that the PrivacyInfo file was added to Amplify-swift  to be compliant with the new guidelines; however, when I try to generate a privacy report for my app including amplify-swift through xCode I receive the following errors:

```
Missing an expected key: 'NSPrivacyCollectedDataTypes'[APPNAME].app/Amplify_Amplify.bundle/PrivacyInfo.xcprivacy
Missing an expected key: 'NSPrivacyCollectedDataTypes' [APPNAME].app/Amplify_AWSCognitoAuthPlugin.bundle/PrivacyInfo.xcprivacy
Missing an expected key: 'NSPrivacyCollectedDataTypes' [APPNAME].app/Amplify_AWSPluginsCore.bundle/PrivacyInfo.xcprivacy
```

Although NSPrivacyCollectedDataTypes looks optional from the docs is seems to be required to be able to generate the report.

Do you have plan for fixing this?

Thank you

### Steps To Reproduce

```swift
Steps to reproduce the behaviour:
1. Include amplify-sdk package dependency in the app
2. generate the archive in xCode Product->archive 
3. open the xCode organizer
4. right click on the archive version number and press 'Generate privacy report'
5. open the archived report PDF and observe
```

### Expected behavior

The privacy report is generated without errors.

### Amplify Framework Version

2.27.0

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5

### CLI version

not relevant

### Xcode version

15.2

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

17.2 (not relevant)

### Device

iPhone (not relevant)

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by 5d** [comment]
Hi @AleCatSS ,

Thanks for opening the issue. Our team will work on reproducing it and will provide an update once we have more information.

**Comment by lawmicha** [comment]
Hi @AleCatSS, we have a PR out that fixes this issue here https://github.com/aws-amplify/amplify-swift/pull/3531

**Comment by 5d** [comment]
Hi @AleCatSS , the PR has been merged and released with version [2.27.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.27.1). 

Closing this issue for now, please feel free to reopen it if the problem persists.

**Comment by romoore** [comment]
I'm still getting emails from Apple that the app is using file metadata APIs. I could easily find that Amplify is using [`contentModificationDateKey`](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393), but doesn't seem to be declaring that usage.  What's the planned timeline for updating the Privacy Manifests for the file timestamp APIs?

https://github.com/aws-amplify/amplify-swift/blob/7d9640a5b7416f02f03463972dc2c6d87b295545/AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Resources/PrivacyInfo.xcprivacy#L20-L30

As of today, it looks like the timestamp APIs are still used. At least in LogRotation (possibly other APIs in other areas).

https://github.com/aws-amplify/amplify-swift/blob/7d9640a5b7416f02f03463972dc2c6d87b295545/AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Persistence/LogRotation.swift#L145

**Comment by romoore** [comment]
> Hi @AleCatSS , the PR has been merged and released with version [2.27.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.27.1).
> 
> Closing this issue for now, please feel free to reopen it if the problem persists.

Opened https://github.com/aws-amplify/amplify-swift/issues/3568 just in case you were mistaken about reopening this one. :)
